src/control/server/server.go

```diff
@@ -190,16 +150,20 @@ func Start(log *logging.LeveledLogger, cfg *Configuration) error {
 				return FaultVfioDisabled
 			}
 
-			if !iommuDetected() {
+			if iommuDisabled {
 				return FaultIommuDisabled
 			}
 		}
 	}
 
 	log.Debugf("automatic NVMe prepare req: %+v", prepReq)
-	if _, err := bdevProvider.Prepare(prepReq); err != nil {
+	prepResp, err := bdevProvider.Prepare(prepReq)
+	if err != nil {
 		log.Errorf("automatic NVMe prepare failed (check configuration?)\n%s", err)
 	}
+	if prepResp.VmdDetected {
+		bdevProvider.EnableVmd()
+	}
```

Comment:
This still seems a little strange... If the bdevProvider detected VMD devices, why wouldn't it just enable it internally? The prepare request has the DisableVMD field, so it seems like the provider should have everything it needs in order to make the decision.

Reply:
if it is a forwarded request then the provider that the request is actually executed on is not the same provider that was used to send the request, enabling VMD on the destination provider (inited by daos_admin) will therefore not be persistent. VMD should be enabled on the source provider instance (which will then persist the setting throughout daos_server lifetime) based on the prepare response generated by the destination provider instance, that's why I'm assigning in server/server.go so that the source provider instance can be updated and the setting persisted.

Reply:
I don't understand this answer. I understand the distinction between local and forwarded requests; I wrote the code. How is setting it in the provider based on the result of the prepare any different from setting it out here in the server package? I'm just pushing to keep logic related to storage management in the storage subpackages, so that future updates don't require wide-ranging changes across the codebase. The server package shouldn't need to change because some detail of VMD parsing or management needed to change.

Looking at the full set of commits, I see that there is still code related to VMD management over in StorageControlService -- I think that this needs to be moved down into bdevProvider as well.